core/src/main/java/bio/terra/pearl/core/service/datarepo/DataRepoExportService.java

```diff
@@ -128,6 +130,34 @@ public void createDataset(StudyEnvironment studyEnv, String datasetName, String
         dataRepoJobService.create(job);
     }
 
+    public void deleteDataset(StudyEnvironment studyEnv, String datasetName) {
+        Dataset dataset = datasetDao.findByDatasetName(datasetName).orElseThrow(() -> new DatasetNotFoundException("Dataset not found."));
+
+        List<DatasetStatus> deletableStatuses = List.of(DatasetStatus.CREATED, DatasetStatus.FAILED);
+        if(!deletableStatuses.contains(dataset.getStatus())) {
+            throw new DatasetDeletionException(String.format("Could not delete dataset %s. Dataset was in %s state, but must be in a terminal state.", datasetName, dataset.getStatus()));
+        }
+
+        JobModel response;
+        try {
+            response = dataRepoClient.deleteDataset(dataset.getTdrDatasetId());
```

Comment:
Was this an intentional decision to favor honoring the request to delete the dataset from TDR over recording that the gesture was made? I could imagine arranging the database transactions such that the gesture is recorded first. Not suggesting a change, just asking the question. It's really a judgement call on what should the state be if there's a failure mid-stream and what's the process for ultimately resolving it.

But also this should probably be `@Transactional`, both for good practice and because there are multiple DAO operations. (This is something that I'm realizing I might have missed in my last set of changes.)

Reply:
I don't think I see the value in recording the gesture first in this case, unless it would be needed for auditing purposes? 
I am also less concerned about being in a broken state where the TDR dataset is deleted but we never recorded the action in our DB. It will be recorded in other places (app logs, TDR logs), and I think the most important thing is prioritizing the fact that a user wanted to destroy the data in TDR, which will be honored.

I will put the DAO operations inside of a txn.